app/src/main/java/org/mozilla/fenix/home/TopPlaceholderViewHolder.kt

```diff
@@ -0,0 +1,20 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.home
+
+import android.view.View
+import org.mozilla.fenix.R
+import org.mozilla.fenix.utils.view.ViewHolder
+
+class TopPlaceholderViewHolder(
```

Comment:
nitpick: I don't understand how this PR addresses the issues from #22402. Could you add a comment explaining what the issue is and how this code addresses it? It doesn't necessarily have to be here, as a class kdoc, but it seemed like an okay place

Sorry to drive by 😅

Reply:
> nitpick: I don't understand how this PR addresses the issues from #22402. Could you add a comment explaining what the issue is and how this code addresses it? It doesn't necessarily have to be here, as a class kdoc, but it seemed like an okay place
> 
> Sorry to drive by 😅

Sure, no worries.  Because we don't know what is being inflated first, we end up having an issue where the first inflated element is the one that is shown in the recyclerview.  

Take a look at `SessionControlView` where I removed the check `homeScreenViewModel.shouldScrollToTopSites`.  This code is how we currently scroll to the top.  The part I removed checks if top sites is inflated.  If it is, then it scrolls to the top (index 0).  However, this is assuming that top sites exists.  The bug is when the user removed all top sites then it won't scroll.

After some investigation, the most simple solution (and most performant) is to just synchronously inflate a top marker.  This way it's the first one that is inflated and it is also the top one.  Recyclerview then keep it in view no matter how many other items are added below it.  Let me know if you have any other questions.  I would suggest you try the current build without top sites, set default to home and cold start Fenix.  You'll see the problem this patch fixes.

Reply:
@rocketsroger Your comment here is good: https://github.com/mozilla-mobile/fenix/pull/22406/files#diff-f6968644d19314ed1385ed1bef14d8a3aaa163999143e11b0449b94292e6be56R48 -> Can we add something similar to the `TopPlaceholderViewHolder` to explain why it exists :)

Reply:
Sure.  I'll add the comments in `TopPlaceholderViewHoder`.  Thanks